### DISCUSSION TITLE
Displaying slingbox name in streamer

### DISCUSSION THREAD

**Discussion by mdehrlich** [GerryDazoo/Slinger#491] [Ideas]
I have added some code to modify the header on the first_buffer.  It replaces the generic "Slingbox"  so the video player shows which of your configs is currently streaming.
the code is inserted at line 521 of the f release.
      
        magic = bytearray(u'Slingbox'.encode('utf-16le'))
        idx= first_buffer.index(magic)
        sourceid = bytearray(15)
        sourceid[:len(name)*2] = bytearray(name.encode('utf-16le'))
        first_buffer[idx:idx+15]= sourceid
        stream_header = first_buffer[0:h264_header_pos]
        print(name,'Stream started at', ts(), len(stream_header), len(first_buffer[h264_header_pos:]))
#Note BedHD at the top line of the SMPlayer

<img width="960" alt="2023-01-13 (1)" src="https://user-images.githubusercontent.com/8422167/212448452-c8330891-1b54-4708-85e5-c28db512e3b4.png">

**Comment by mdehrlich** [comment]
In VLC
![Screenshot_20230113-231720](https://user-images.githubusercontent.com/8422167/212450504-1ed4c0ed-5938-4836-b7e0-a31277fadae5.png)

**Comment by GerryDazoo** [comment]
I was going to get around to this some day. THX for the code, I'll look at putting it in the next release. (Coming soon)

**Comment by Toei79** [comment]
wish can help in that way! pretty sure you get more people involved.

**Comment by mdehrlich** [comment]
Some revisions to deal with names of 8 characters, and only insert the name if 8 or less characters.

     if len(name) <= 8: # Slingbox name must not be more than 8 characters to display as stream name
            magic = bytearray(u'Slingbox'.encode('utf-16le'))
            idx= first_buffer.index(magic) # find the streamname in the header (idx might always be 214?)
            sourceid = bytearray(16)
            sourceid[:len(name)*2] = bytearray(name.encode('utf-16le'))
            first_buffer[idx:idx+16]= sourceid
        stream_header = first_buffer[0:h264_header_pos]

**Reply by GerryDazoo** [reply]
Already fixed that. :-). I decided to truncate the name instead of not using it at all

**Reply by mdehrlich** [reply]
thumbs up
